src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginEntry/FileOriginEntryViewModel.cs

```diff
@@ -83,5 +98,23 @@ public class FileOriginEntryViewModel : AViewModel<IFileOriginEntryViewModel>, I
             .Merge(interval.Select(_ => LastInstalledDate))
             .Select(date => date.Equals(DateTime.MinValue) ? "-" : date.Humanize())
             .ToProperty(this, vm => vm.DisplayLastInstalledDate, scheduler: RxApp.MainThreadScheduler);
+
+        ViewModCommand = ReactiveCommand.Create<NavigationInformation>(info =>
```

Comment:
You should make the commands settable and set them from `FileOriginsPageViewModel.cs` when FileOriginEntryVM is created in the transform. This allows having access to everything in FileOriginsPageVM and de duplicating functionality such as Add

Reply:
I moved out initializing `viewModCommand` to parent for now, since it's the same for all items, but can you elaborate a bit more on this tomorrow?

I can technically pass an `Add` method to the entry, but then that's just inverting control, so rather than VM calling Entry, the Entry calls VM. I'm not sure what's gained in that situation, since it's just swapping the two. The actual code to add mod is a single method call, so there's not much logic there.